packages/blaze-query/src/blockfrost.ts

```diff
@@ -584,6 +590,103 @@ export class Blockfrost implements Provider {
       Array.from(evaledRedeemers).map((x) => x.toCore()),
     );
   }
+
+  private async getScriptRef(scriptHash: ScriptHash): Promise<Script> {
+    const typeQuery = `/scripts/${scriptHash}`;
+    const typeJson = await fetch(`${this.url}${typeQuery}`, {
+      headers: this.headers(),
+    }).then((resp) => resp.json());
+
+    if (!typeJson) {
+      throw new Error("getScriptRef: Could not parse response json");
+    }
+
+    const typeResponse = typeJson as BlockfrostResponse<{
+      type: "timelock" | "plutusV1" | "plutusV2";
+    }>;
+
+    if ("message" in typeResponse) {
+      throw new Error(
+        `getScriptRef: Blockfrost threw "${typeResponse.message}"`,
+      );
+    }
+
+    const type = typeResponse.type;
+    if (type == "timelock") {
+      throw new Error("getScriptRef: Native scripts are not yet supported.");
+    }
+
+    const cborQuery = `/scripts/${scriptHash}/cbor`;
+    const cborJson = await fetch(`${this.url}${cborQuery}`, {
```

Comment:
@EzePze  Wouldn't this single call to BF be enough for this method ? According to [the docs](https://blockfrost.dev/api/script-cbor), if the scriptHash is for a nativescript then the response will be `null`:

> CBOR contents of the plutus script, null for `timelocks`

In this case we might not need the other call on line 595 at all.

Reply:
True, but we need to know what the plutus version is when constructing the serialised script on lines 640-645

Reply:
I see now. Thanks